### ISSUE TITLE
User search should order prefix matches first

### ISSUE THREAD

**Issue by alya** [zulip/zulip-mobile#4869]
At present, user search results are shown with all matches in alphabetical order. This doesn't match what the user is trying to do, which is usually to put in the *start* of someone's name. It looks especially "broken" when the user first start typing, as it's unclear why these particular names are being shown, or why the name list didn't seem to change (or change much) when the first letter was typed.

We should change the ordering to show prefix matches with any part of the name above other results, the way we currently do in the left sidebar of the web app. It should be fine to just use <space> as the separator, or match whatever implementation we have in the web app.

**Comment by thisisnitish** [comment]
@aly!, I tried a lot with this issue. Could you please explain a little bit more. I still not able to figure out where to start. Thanks 🙂

**Comment by alya** [comment]
Hi @thisisnitish , please post any specific questions you have, as it's hard to help with such a general request. See [here](https://jvns.ca/blog/good-questions/) for some tips on asking good questions.

**Comment by thisisnitish** [comment]
Thanks @alya!, since we both have worked on some issues, so I thought you will understand. 
> See [here](https://jvns.ca/blog/good-questions/) for some tips on asking good questions.

Sure I'll check it once and I will get back to you. Thanks 😊

**Comment by thisisnitish** [comment]
@alya! Till now, I've observed is, searching of user doesn't starts from the first character it starts from the second character. But in our web app, searching starts from the beginning itself. Correct me if I'm wrong. Thanks 🙂

**Comment by alya** [comment]
@thisisnitish Yes, there is a separate issue for changing that behavior on mobile: #4872.

**Comment by gnprice** [comment]
Edited title to make explicit that this is about the user-search screen (the one you get from the "New PM" button), and not the typeahead for @-mentions. (The same thing should apply there, but it's a separate implementation.)
